src/command_modules/azure-cli-vm/tests/test_vm_commands.py

```diff
@@ -392,6 +392,50 @@ def body(self):
         ])
 
 
+class VMAttachDisksOnCreate(ScenarioTest):
+
+    @ResourceGroupPreparer()
+    def test_vm_create_by_attach_os_and_data_disks(self, resource_group):
+        # the testing below follow a real custom's workflow requiring the support of attaching data disks on create
+
+        # creating a vm
+        self.cmd('vm create -g {} -n vm1 --image centos --admin-username centosadmin --admin-password testPassword0 --authentication-type password --data-disk-sizes-gb 2'.format(resource_group))
+        result = self.cmd('vm show -g {} -n vm1'.format(resource_group)).get_output_in_json()
+        origin_os_disk_name = result['storageProfile']['osDisk']['name']
+        origin_data_disk_name = result['storageProfile']['dataDisks'][0]['name']
+
+        # snapshot the os & data disks
+        os_snapshot = 'oSnapshot'
+        os_disk = 'sDisk'
+        data_snapshot = 'dSnapshot'
+        data_disk = 'dDisk'
+        self.cmd('snapshot create -g {} -n {} --source {}'.format(resource_group, os_snapshot, origin_os_disk_name))
+        self.cmd('disk create -g {} -n {} --source {}'.format(resource_group, os_disk, os_snapshot))
+        self.cmd('snapshot create -g {} -n {} --source {}'.format(resource_group, data_snapshot, origin_data_disk_name))
+        self.cmd('disk create -g {} -n {} --source {}'.format(resource_group, data_disk, data_snapshot))
+
+        # rebuild a new vm
+        self.cmd('vm create -g {} -n vm2 --attach-os-disk {} --attach-data-disks {} --os-type linux'.format(resource_group, os_disk, data_disk),
+                 checks=[JMESPathCheckV2('powerState', 'VM running')])
+
+    @ResourceGroupPreparer()
+    def test_vm_create_by_attach_unmanaged_os_and_data_disks(self, resource_group):
+        # creating a vm
+        self.cmd('vm create -g {} -n vm1 --use-unmanaged-disk --image centos --admin-username centosadmin --admin-password testPassword0 --authentication-type password'.format(resource_group))
+        self.cmd('vm unmanaged-disk attach -g {} --vm-name vm1 --new --size-gb 2'.format(resource_group))
+        result = self.cmd('vm show -g {} -n vm1'.format(resource_group)).get_output_in_json()
+        os_disk_vhd = result['storageProfile']['osDisk']['vhd']['uri']
+        data_disk_vhd = result['storageProfile']['dataDisks'][0]['vhd']['uri']
+
+        # delete the vm to end vhd's leases so they can be used to create a new vm through attaching
+        self.cmd('vm deallocate -g {} -n vm1'.format(resource_group))
+        self.cmd('vm delete -g {} -n vm1 -y'.format(resource_group))
+
+        # rebuild a new vm
+        self.cmd('vm create -g {} -n vm2 --attach-os-disk {} --attach-data-disks {} --os-type linux --use-unmanaged-disk'.format(
```

Comment:
It looks like all the tests are either creating new data disks or attaching existing ones, but the code seems to suggest you can do both.  If that is supported, consider adding a --data-disk-sizes-gb entry to this test to make sure the service doesn't throw up.